Tasks/Common/AzureRmDeploy-common/azure-arm-rest/azure-arm-app-service.ts

```diff
@@ -244,16 +244,19 @@ export class AzureAppService {
     public async patchApplicationSettingsSlot(addProperties: any): Promise<any> {
         var appSettingsSlotSettings = await this.getSlotConfigurationNames();
         let appSettingNames = appSettingsSlotSettings.properties.appSettingNames;
+        if(!appSettingNames)
+        {
+            appSettingsSlotSettings.properties.appSettingNames = [];
```

Comment:
Can we instead add null check inside for loop. This will ensure we create array only if there are some key-value pairs inside object parsed to the function.